sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,150 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../testHelpers";
+import {
+  createFakeInstitutionLocation,
+  saveFakeStudent,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { Institution, InstitutionLocation } from "@sims/sims-db";
+
+describe("ApplicationInstitutionsController(e2e)-getApplicationDetails", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeCLocation: InstitutionLocation;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // College C.
+    const { institution: collegeC } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+  });
+
+  it(
+    "Should get the student application details when student has a submitted application " +
+      "for the institution (application with location id saved).",
+    async () => {
+      // Arrange
+      // Student has a submitted application to the institution.
+      const student = await saveFakeStudent(appDataSource);
```

Comment:
Is there a benefit on creating the student here instead of allowing the `saveFakeStudent` do it?

Reply:
Having the student creation in the beforeall, fails the last test case. So to ignore the conflict, creating it seperately.

Reply:
I am not sure how we end up having the `beforeall` in this conversation. The only need that I recommended was leaving the student creation to the `saveFakeApplication`.
Please check if the below code makes sense or maybe I am missing something.
```ts
       // Create new application.
      const savedApplication = await saveFakeApplication(appDataSource, {
        institution: collegeF,
        institutionLocation: collegeFLocation,
      });
      const student = savedApplication.student;

      const endpoint = `/institutions/application/${savedApplication.id}/student/${student.id}`;
      const institutionUserToken = await getInstitutionToken(
        InstitutionTokenTypes.CollegeFUser,
      );
```